examples/allocation/zig/testdata/greet.zig

```diff
@@ -0,0 +1,79 @@
+const std = @import("std");
+const allocator = std.heap.page_allocator;
+
+// main is required for Zig to compile to enable WASI.
+pub fn main() anyerror!void {
+    return;
+}
+
+extern "env" fn log(ptr: u32, size: u32) void;
```

Comment:
Although `u32` is technically correct, you're accepting a pointer and a pointer-size, so a more correct function signature would be something like: `extern "env" fn log(ptr: [*]const u8, size: usize) void;`.

Then in your `_log` function it can be further simplified to:
```zig
pub fn _log(message: []const u8) void {
    log(message.ptr, message.len);
}
```

Reply:
Awesome.
https://github.com/tetratelabs/wazero/pull/741/commits/b191b90928b9de4c53cd4f485daf0d71658fcba2